matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/create/CreateRoomBodyBuilder.kt

```diff
@@ -181,7 +181,7 @@ internal class CreateRoomBodyBuilder @Inject constructor(
                 params.invite3pids.isEmpty() &&
                 params.invitedUserIds.isNotEmpty() &&
                 params.invitedUserIds.let { userIds ->
-            val keys =  olmMachineProvider.olmMachine.getUserDevicesMap(userIds)
+            val keys =  olmMachineProvider.olmMachine.ensureUserDevicesMap(userIds, forceDownload = false)
```

Comment:
Do we want to switch to a internal API here? Now that `downloadKeys()` does the right thing, why not continue using it?

Reply:
We try to avoid to inject services into internal classes.

Reply:
Why is that? The old code seems to have used it just fine:
```kotlin
// deviceListManager.downloadKeys(userIds, forceDownload = false)
```

Using internal APIs only serves to tightly couple components which shouldn't be coupled and prevents easy changes in the future. It also increases the potential of misusing the components and increasing breakage.

Reply:
Yeah i would need to create a DeviceListManager equivalent for rust that would be imported by both cryptoService and CreateRoomBodyBuilder (and maybe others)

Reply:
In that case, I guess, it's fine to use the Olm machine here.